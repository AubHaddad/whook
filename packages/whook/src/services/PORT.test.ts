import initPORT from './PORT';
import initImporter from './importer';
import type { PortFinderModule } from './PORT';

describe('initPORT', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should use the env port first', async () => {
    const importer = await initImporter<PortFinderModule>({ log });
    const port = await initPORT({
      ENV: { PORT: '1337' },
      importer,
      log,
    });

    expect({
      port,
    }).toMatchInlineSnapshot(`
      Object {
        "port": 1337,
      }
    `);
    expect({
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });

  it('should find a port by itself if no env port', async () => {
    const importer = await initImporter<PortFinderModule>({ log });
    const port = await initPORT({
      importer,
      log,
    });

    expect(port).toBeGreaterThan(0);
    expect({
      logCalls: log.mock.calls
        .filter((args) => 'stack' !== args[0])
        .map(([arg1, arg2, ...args]) => {
          return [arg1, arg2.replace(/port (\d+)/, 'port ${PORT}'), ...args];
        }),
    }).toMatchSnapshot();
  });
});
