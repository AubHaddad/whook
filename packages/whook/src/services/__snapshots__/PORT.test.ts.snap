// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initPORT should find a port by itself if no env port 1`] = `
{
  "logCalls": [
    [
      "debug",
      "🛂 - Initializing the importer!",
    ],
    [
      "debug",
      "🏭 - Initializing the PORT service.",
    ],
    [
      "debug",
      "🛂 - Dynamic import of "portfinder".",
    ],
    [
      "warning",
      "✔ - Found a free port "8000"",
    ],
  ],
}
`;

exports[`initPORT should use the env port first 2`] = `
{
  "logCalls": [
    [
      "debug",
      "🛂 - Initializing the importer!",
    ],
    [
      "debug",
      "🏭 - Initializing the PORT service.",
    ],
    [
      "warning",
      "♻️ - Using ENV port "1337"",
    ],
  ],
}
`;
